src/libasr/pass/pass_list_concat.cpp

```diff
@@ -0,0 +1,247 @@
+#include <libasr/asr.h>
+#include <libasr/containers.h>
+#include <libasr/exception.h>
+#include <libasr/asr_utils.h>
+#include <libasr/asr_verify.h>
+#include <libasr/pass/pass_list_concat.h>
+#include <libasr/pass/pass_utils.h>
+
+#include <vector>
+#include <utility>
+
+
+namespace LFortran {
+
+using ASR::down_cast;
+
+/*
+    This ASR pass replaces ListConcat with a FunctionCall where
+    the two lists are passed as arguments to the function.
+    Then, the function returns the concatenated list as a result.
+
+    Converts:
+        x = i + j // lists
+
+    to:
+        x = _lcompilers_list_concat(i, j)
+ */
+
+
+class ReplaceListConcat : public ASR::BaseExprReplacer<ReplaceListConcat>
+{
+private:
+
+    Allocator& al;
+    ASR::TranslationUnit_t &unit;
+    bool is_list_concat_defined;
+
+public:
+    ReplaceListConcat(Allocator &al_, ASR::TranslationUnit_t &unit_,
+                      bool is_list_concat_defined_) :
+        al(al_), unit(unit_), is_list_concat_defined(is_list_concat_defined_)
+        { }
+
+    ASR::symbol_t* create_concat_function(Location& loc,
+            SymbolTable*& global_scope, ASR::ttype_t* list_type) {
+        if (is_list_concat_defined) {
+            return global_scope->get_symbol("_lcompilers_list_concat");
```

Comment:
What if the user created the function "_lcompilers_list_concat"

Reply:
In that case, I think instead of using the flag, we should store the function name. This might helo us.

Reply:
Done.